plugins/languages/clangd-extensions.nix

```diff
@@ -20,157 +68,171 @@ in
+          right_align = rightAlign;
+          right_align_padding = rightAlignPadding;
+          inherit
+            highlight
+            priority
+            ;
+        };
+        ast = with ast; {
+          role_icons = with roleIcons; {
+            inherit
+              type
+              declaration
+              expression
+              statement
+              specifier
+              ;
+            "template argument" = templateArgument;
           };
-          memory_usage = {
-            inherit (cfg.extensions.memoryUsage) border;
+          kind_icons = with kindIcons; {
+            Compound = compound;
+            Recovery = recovery;
+            TranslationUnit = translationUnit;
+            PackExpansion = packExpansion;
+            TemplateTypeParm = templateTypeParm;
+            TemplateTemplateParm = templateTemplateParm;
+            TemplateParamObject = templateParamObject;
           };
-          symbol_info = {
-            inherit (cfg.extensions.symbolInfo) border;
+          highlights = with highlights; {
+            inherit detail;
           };
         };
-      };
-    in
-      mkIf cfg.enable {
-        extraPackages = [cfg.server.package];
-        extraPlugins = [cfg.package];
-
-        plugins.clangd-extensions.server.extraOptions = mkIf cfg.enableOffsetEncodingWorkaround {
-          capabilities = {__raw = "__clangdCaps";};
+        memory_usage = with memoryUsage; {
+          inherit border;
+        };
+        symbol_info = with symbolInfo; {
+          inherit border;
         };
+      }
+      // cfg.extraOptions;
+  in
+    mkIf cfg.enable {
+      warnings = optional (!config.plugins.lsp.enable) ''
+        You have enabled `clangd-extensions` but not the lsp (`plugins.lsp`).
+        You should set `plugins.lsp.enable = true` to make use of the clangd-extensions' features.
+      '';
 
-        plugins.lsp.postConfig = let
-          extraCaps =
-            if cfg.enableOffsetEncodingWorkaround
-            then ''
-              local __clangdCaps = vim.lsp.protocol.make_client_capabilities()
-              __clangdCaps.offsetEncoding = { "utf-16" }
-            ''
-            else "";
-        in ''
-          ${extraCaps}
-          require("clangd_extensions").setup(${helpers.toLuaObject setupOptions})
-        '';
+      plugins.lsp.servers.clangd.extraOptions = mkIf cfg.enableOffsetEncodingWorkaround {
+        capabilities = {__raw = "__clangdCaps";};
       };
-  }
+
+      extraPlugins = [cfg.package];
+
+      # Enable the clangd language server
+      plugins.lsp.servers.clangd.enable = true;
+
+      plugins.lsp.postConfig = let
+        extraCaps =
+          if cfg.enableOffsetEncodingWorkaround
+          then ''
+            local __clangdCaps = vim.lsp.protocol.make_client_capabilities()
+            __clangdCaps.offsetEncoding = { "utf-16" }
+          ''
+          else "";
+      in ''
+        ${extraCaps}
```

Comment:
This define the __clangdCaps in the postConfig, but it is used in the LSP config itself. I think you need to do that in the preConfig

Reply:
Fixed